tools/chatgpt/chatgpt.xml

```diff
@@ -0,0 +1,109 @@
+<tool id="chatgpt" name="chatGPT" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" python_template_version="3.5" profile="21.05">
+    <description>Integrating OpenAI's ChatGPT into Galaxy</description>
+    <macros>
+        <token name="@TOOL_VERSION@">2024</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="3.12.4">python</requirement>
+        <requirement type="package" version="1.35.13">openai</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+#set count = 1
+#set LINK_LIST = ""
+#for $input in $context:
+    #set LINK = "input" + str($count) + "." + $input.ext
+    ln -s "$input" "$LINK" &&
+    #if count == 1
+        #set LINK_LIST = $LINK
+    #else
+        #set LINK_LIST = $LINK_LIST + "," + $LINK
+    #end if
+    #set count = $count + 1
+#end for
+
+python "$__tool_directory__/chatgpt.py"
+"$LINK_LIST"
+"$question"
+"$model"
+"$openai_api_key"
+    ]]></command>
+    <configfiles>
+        <configfile name="openai_api_key"><![CDATA[
+$__user__.extra_preferences.get('chatgpt|api_key', "")
+        ]]></configfile>
+    </configfiles>
+    <inputs>
+        <param name="context" type="data" multiple="true" optional="false" format="doc,docx,html,json,pdf,txt,jpg,jpeg,png,webp,gif" label="Context"/>
+        <param name="question" type="text" optional="false" label="Question" help="Question about the text provided"/>
+        <param name="model" type="select" optional="false" label="Model" help="Select the model you want to use">
+            <option value="gpt-4o-mini">Affordable and intelligent small model for fast, lightweight tasks (gpt-4o-mini)</option>
+            <option value="gpt-4o">High-intelligence flagship model for complex, multi-step tasks (gpt-4o)</option>
```

Comment:
How would a multi-step-task look like?

Reply:
What do you mean by multi-step-task?

Reply:
You wrote it ... just read the sentence ...

Reply:
Well, this is part of the OpenAI API documentation, and I assume it means that the model can handle questions requiring a complex sequence of steps, completing them accurately.

Reply:
And do you think it makes sense in the Galaxy context?

Reply:
What I think it is describing seems to mean that the model can handle multiple tasks or jobs within a single prompt, processing them sequentially and then providing all the results in one comprehensive reply.